### PR TITLE
python311Packages.debugpy: 1.6.7.post1 -> 1.8.0

### PR DESCRIPTION
**Pull request by gepbird**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by gepbird** [comment]
Couldn't build old 1.6.7.post1 version, since it doesn't support python 3.11, so I tried to bump it to 1.8.0 which supports it.

While building the package, tests failed with the error message `[gw1] node down: Not properly terminated`.
I saw this error message on debugpy's previous CI builds too, but their 1.8.0 CI passed. I'd appriciate any help with debugging this :)

**Comment by gepbird** [comment]
@kira-bruneau do you have any idea why did gw crash?

**Comment by kira-bruneau** [comment]
Oh I tried updating about a week ago and ran into the same problem, but wasnt quite sure what was going on yet.

We could disable the test that's failing but I'd rather figure out why its failing. Forgot to get back to it, but I can take a look at it now!

File: pkgs/development/python-modules/debugpy/default.nix [hunk 1]
```diff
@@ -50,13 +47,6 @@ buildPythonPackage rec {
     # To avoid this issue, debugpy should be installed using python.withPackages:
     # python.withPackages (ps: with ps; [ debugpy ])
     ./fix-test-pythonpath.patch
-
```
Comment (kira-bruneau) [inline pkgs/development/python-modules/debugpy/default.nix:53]: Should also remove `fetchpath` as an argument now that we no longer need it.

**Review by kira-bruneau — CHANGES_REQUESTED** [review]
Hmmm, this is a bit of a weird one. It looks like this is failing because of a deprecation warning:
```
Traceback (most recent call last):
  File "/nix/store/lx8vhp4fxclp494svlfis3sb2g8z4l9h-python3-3.10.12/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/nix/store/lx8vhp4fxclp494svlfis3sb2g8z4l9h-python3-3.10.12/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/nix/store/wbk5b683rzzj0xqb90zv1blx1j2zrpvr-python3.10-debugpy-1.7.0/lib/python3.10/site-packages/debugpy/adapter/../../debugpy/launcher/../../debugpy/__main__.py", line 39, in <module>
    cli.main()
  File "/nix/store/wbk5b683rzzj0xqb90zv1blx1j2zrpvr-python3.10-debugpy-1.7.0/lib/python3.10/site-packages/debugpy/adapter/../../debugpy/launcher/../../debugpy/../debugpy/server/cli.py", line 430, in main
    run()
  File "/nix/store/wbk5b683rzzj0xqb90zv1blx1j2zrpvr-python3.10-debugpy-1.7.0/lib/python3.10/site-packages/debugpy/adapter/../../debugpy/launcher/../../debugpy/../debugpy/server/cli.py", line 284, in run_file
    runpy.run_path(target, run_name="__main__")
  File "/nix/store/wbk5b683rzzj0xqb90zv1blx1j2zrpvr-python3.10-debugpy-1.7.0/lib/python3.10/site-packages/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_runpy.py", line 321, in run_path
    return _run_module_code(code, init_globals, run_name,
  File "/nix/store/wbk5b683rzzj0xqb90zv1blx1j2zrpvr-python3.10-debugpy-1.7.0/lib/python3.10/site-packages/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_runpy.py", line 135, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File "/nix/store/wbk5b683rzzj0xqb90zv1blx1j2zrpvr-python3.10-debugpy-1.7.0/lib/python3.10/site-packages/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_runpy.py", line 124, in _run_code
    exec(code, run_globals)
  File "/build/pytest-of-nixbld/pytest-0/popen-gw9/test_gevent_program_launch_0/code_to_debug.py", line 7, in <module>
    monkey.patch_all()
  File "/nix/store/r9jk4ynzh95jw9xijcf89d66a4kqc6dc-python3.10-gevent-22.10.2/lib/python3.10/site-packages/gevent/monkey.py", line 1253, in patch_all
    from gevent import events
  File "/nix/store/r9jk4ynzh95jw9xijcf89d66a4kqc6dc-python3.10-gevent-22.10.2/lib/python3.10/site-packages/gevent/events.py", line 74, in <module>
    from pkg_resources import iter_entry_points
  File "/nix/store/4mm6znbhkrxm02q9gda7xwfqj7dnc53r-python3.10-setuptools-68.0.0/lib/python3.10/site-packages/pkg_resources/__init__.py", line 118, in <module>
    warnings.warn(
DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
```

There was an issue open for this at https://github.com/microsoft/debugpy/issues/1375, which should've been fixed by https://github.com/microsoft/debugpy/pull/1374? I have a feeling that their workaround is designed for their specific CI environment, and still fails under a nix build.

I think it'd make sense to just skip the test and leave a note of it, something like this:
```nix
  disabledTests = [
    # DeprecationWarning: pkg_resources is deprecated as an API
    # Supposedly fixed in https://github.com/microsoft/debugpy/pull/1374,
    # but still fails for a nix build
    "test_gevent"
  ];
```

Oh also:
- ~It looks like we don't need to disable `test_flask_breakpoint_multiproc` anymore~
- Now that `fetchpatch` is no longer needed, it should be removed as an argument at the top of the file

**Comment by kira-bruneau** [comment]
Oh oops looks like we still need to disable `test_flask_breakpoint_multiproc`!

**Comment by gepbird** [comment]
@kira-bruneau Thanks for your deep research! I'm disabling that test.

When checking the diff between 1.6.7.post1 and 1.8.0, I saw a lot of setuptools removal, so I removed it from this package. After double-checking it, looks like they only removed it from tests, so I'm adding back setuptools.

**Comment by kira-bruneau** [comment]
Result of `nixpkgs-review pr 255379` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>home-assistant-component-tests.debugpy</li>
    <li>python310Packages.debugpy</li>
    <li>python310Packages.debugpy.dist</li>
    <li>python311Packages.debugpy</li>
    <li>python311Packages.debugpy.dist</li>
    <li>vscode-extensions.ms-python.python</li>
  </ul>
</details>

**Review by kira-bruneau — APPROVED** [review]

**Comment by kira-bruneau** [comment]
Thanks @gepbird!!

**Comment by gepbird** [comment]
@kira-bruneau I was testing if it works by running `nix-shell -I nixpkgs=/path/to/my/nixpkgs/fork -p python311Packages.debugpy --run "python -m debugpy"`, and it throws a new warning (old version has no warnings), is this a packaging issue?:
```
0.01s - Debugger warning: It seems that frozen modules are being used, which may
0.00s - make the debugger miss breakpoints. Please pass -Xfrozen_modules=off
0.00s - to python to disable frozen modules.
0.00s - Note: Debugging will proceed. Set PYDEVD_DISABLE_FILE_VALIDATION=1 to disable this validation.
debugpy 1.8.0
See https://aka.ms/debugpy for documentation.
```

**Comment by gepbird** [comment]
Another thing, that is maybe unrelated to this PR and should be an issue but: I'm pretty sure a few days ago I could install debugpy like this:
```nix
environment.systemPackages = with pkgs; [
  python3
  python310Packages.debugpy
];
```
and run it with `python -m debugpy`. But this doesn't work now: `/home/user/.nix-profile/bin/python: No module named debugpy`.

Currently I can use debugpy like this, but putting every python package in the same file is inconvenient:
```
environment.systemPackages = with pkgs; [
  (python311.withPackages(ps: with ps; [ debugpy ]))
];
```

**Pull request merged**